packages/backend/src/apps/paysg/auth/verify-credentials.ts

```diff
@@ -11,6 +11,28 @@ export default async function verifyCredentials(
     throw new Error('Invalid PaySG API key')
   }
 
+  const paymentServiceId = authData.paymentServiceId as string
+
+  if (!paymentServiceId || typeof paymentServiceId !== 'string') {
+    throw new Error('Payment Service ID must be set')
+  }
+
```

Comment:
Nit: since we're not doing paysg URL parsing yet, we can add this as a stopgap.

Left the 52 magic number inside code block for brevity; can move to a const or something.

```suggestion

if (
    !paymentServiceId.startsWith('payment_service_') ||
    paymentServiceId.length != 52 // prefix + UUID
  ) {
    throw new Error('Invalid Payment Service ID')
  } 

```

Reply:
Actually we can park this inside `addAuthHeader` to be more general